controllers/city_controller_test.go

```diff
@@ -0,0 +1,130 @@
+package controllers
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/NikolNikolaeva/project_weather/generated/dao/model"
+	"github.com/NikolNikolaeva/project_weather/mocks"
+)
+
+func Test_GetCityById(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockCityRepo := repositories.NewMockCityRepo(ctrl)
+	controller := NewCityController(mockCityRepo)
+
+	testID := "1"
+	testCity := &model.City{ID: testID, Name: "Sofia", Country: "Bulgaria"}
+
+	mockCityRepo.EXPECT().FindCityByID(testID).Return(testCity, nil)
+
+	app := fiber.New()
+	app.Get("/cities/:id", controller.GetCityById)
+
+	req := httptest.NewRequest(http.MethodGet, "/cities/1", nil)
+	resp, err := app.Test(req)
+
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+}
```

Comment:
Add assertion for the model also, smt like `assert.Equal(t, testCity, resp.Body...City)